tapgarden/batch.go

```diff
@@ -141,6 +141,18 @@ func (m *MintingBatch) validateGroupAnchor(s *Seedling) error {
 			*s.GroupAnchor)
 	}
 
+	// The decimal display of the seedling must match that of the group
+	// anchor. We already validated the seedling metadata, so we don't care
+	// if the value is explicit or if the metadata is JSON, but we must
+	// compute the same value for both assets.
+	_, seedlingDecDisplay, _ := s.Meta.GetDecDisplay()
+	_, anchorDecDisplay, _ := anchor.Meta.GetDecDisplay()
```

Comment:
do we always assume that the group anchor is in the same batch?

Reply:
In `tapgarden.Seedling` (which is a bit like the batch minting request) the group anchor is optional. But if set, then an attempt is made to retrieved  the anchor from the batch. So, as far as I can tell, the group anchor doesn't have to be in the batch.

Reply:
a) Should we also check if `anchor != nil` here?
b) So if the anchor is not in the batch, do we skip the decimal display check?

Reply:
In this method we've already retrieved the anchor using `anchor, ok := m.Seedlings[*s.GroupAnchor]`. And we return an error if it's missing. We should never really be adding a `nil` entry into `m.Seedlings`, but I could double check at this point. I will add that check.

Reply:
I've gone with
```
	if anchor == nil || !ok {
		return fmt.Errorf("group anchor %v not present in batch",
			s.GroupAnchor)
	}
```

Reply:
> a) Should we also check if `anchor != nil` here?
> b) So if the anchor is not in the batch, do we skip the decimal display check?

(b) is covered here: https://github.com/lightninglabs/taproot-assets/pull/952#discussion_r1643136426